test_unstructured/partition/test_html.py

```diff
@@ -906,4 +901,187 @@ def __init__(self, text: str, status_code: int, headers: dict[str, str] = {}):
+
+    # -- .detection_origin -----------------------
+
+    @pytest.mark.parametrize("detection_origin", ["html", None])
+    def it_knows_the_caller_provided_detection_origin(
+        self, detection_origin: str | None, opts_args: dict[str, Any]
+    ):
+        opts_args["detection_origin"] = detection_origin
+        opts = HtmlPartitionerOptions(**opts_args)
+
+        assert opts.detection_origin == detection_origin
+
+    # -- .encoding -------------------------------
+
+    @pytest.mark.parametrize("encoding", ["utf-8", None])
+    def it_knows_the_caller_provided_encoding(
+        self, encoding: str | None, opts_args: dict[str, Any]
+    ):
+        opts_args["encoding"] = encoding
+        opts = HtmlPartitionerOptions(**opts_args)
+
+        assert opts.encoding == encoding
+
+    # -- .html_str -------------------------------
+
+    def it_gets_the_HTML_from_the_file_path_when_one_is_provided(self, opts_args: dict[str, Any]):
+        file_path = example_doc_path("example-10k-1p.html")
+        opts_args["file_path"] = file_path
+        opts = HtmlPartitionerOptions(**opts_args)
+
+        html_str = opts.html_str
+
+        assert isinstance(html_str, str)
+        assert html_str == read_txt_file(file_path)[1]
+
+    def and_it_gets_the_HTML_from_the_file_like_object_when_one_is_provided(
+        self, opts_args: dict[str, Any]
+    ):
+        file_path = example_doc_path("example-10k-1p.html")
+        with open(file_path, "rb") as f:
+            file = io.BytesIO(f.read())
+        opts_args["file"] = file
+        opts = HtmlPartitionerOptions(**opts_args)
+
+        html_str = opts.html_str
+
+        assert isinstance(html_str, str)
+        assert html_str == read_txt_file(file_path)[1]
+
+    def and_it_uses_the_HTML_in_the_text_argument_when_that_is_provided(
+        self, opts_args: dict[str, Any]
+    ):
+        opts_args["text"] = "<html><body><p>Hello World!</p></body></html>"
+        opts = HtmlPartitionerOptions(**opts_args)
+
+        assert opts.html_str == "<html><body><p>Hello World!</p></body></html>"
+
+    def and_it_gets_the_HTML_from_the_url_when_one_is_provided(
+        self, requests_get_: Mock, opts_args: dict[str, Any]
+    ):
+        requests_get_.return_value = FakeResponse(
+            text="<html><body><p>I just flew over the internet!</p></body></html>",
+            status_code=200,
+            headers={"Content-Type": "text/html"},
+        )
+        opts_args["url"] = "https://insta.tweet.face.org"
+        opts = HtmlPartitionerOptions(**opts_args)
+
+        assert opts.html_str == "<html><body><p>I just flew over the internet!</p></body></html>"
+
+    def but_it_raises_when_no_path_or_file_or_text_or_url_was_provided(
+        self, opts_args: dict[str, Any]
+    ):
+        opts = HtmlPartitionerOptions(**opts_args)
+
+        with pytest.raises(ValueError, match="Exactly one of filename, file, text, or url must be"):
+            opts.html_str
+
```

Comment:
Should the `test_partition_html_raises_with_too_many_specified` that was removed be moved here?

Reply:
The "raises_with_too_many_specified" behavior is no longer present. It just uses the first one it finds in `filename`, `file`, `text`, `url` order.

If someone can show me support cases where folks are providing more than one and then being confused about the behavior we can add that back in, but my assumption is it was just speculation that such a behavior would be worthwhile and I don't believe it is.